membership-common/src/main/scala/com/gu/memsub/subsv2/Plan.scala

```diff
@@ -122,178 +108,126 @@ case object Years extends UpToPeriodsType {
+          s"period =${zBillingPeriod} specificBillingPeriod=${specificBillingPeriod} uptoPeriodsType=${upToPeriodsType}, uptoPeriods=${upToPeriods}",
+        )
+    }
+
 }
 
 /** Low level model of a rate plan, as it appears on a subscription in Zuora
   */
-case class SubscriptionZuoraPlan(
+case class RatePlan(
     id: RatePlanId,
     productRatePlanId: ProductRatePlanId,
     productName: String,
     lastChangeType: Option[String],
     features: List[Feature],
     chargedThroughDate: Option[LocalDate],
-    charges: NonEmptyList[ZuoraCharge],
+    ratePlanCharges: NonEmptyList[RatePlanCharge],
     start: LocalDate,
     end: LocalDate,
 ) {
-  def price = charges.list.toList.flatMap { charge =>
-    charge.pricing.prices.map(_.amount)
-  }.sum
-}
 
-/** Low level model of a product rate plan, as it appears in the Zuora product catalog
-  */
-case class CatalogZuoraPlan(
-    id: ProductRatePlanId,
-    name: String,
-    description: String,
-    productId: ProductId,
-    saving: Option[String],
-    charges: List[ZuoraCharge],
-    benefits: Map[ProductRatePlanChargeId, Benefit],
-    status: Status,
-    frontendId: Option[FrontendId],
-    private val productTypeOption: Option[String],
-) {
-  lazy val productType = productTypeOption.getOrElse(throw new RuntimeException("Product type is undefined for plan: " + name))
-}
-
-sealed trait FrontendId {
-  def name: String
-}
-object FrontendId {
-
-  case object OneYear extends FrontendId { val name = "OneYear" }
-  case object ThreeMonths extends FrontendId { val name = "ThreeMonths" }
-  case object Monthly extends FrontendId { val name = "Monthly" }
-  case object Quarterly extends FrontendId { val name = "Quarterly" }
-  case object Yearly extends FrontendId { val name = "Yearly" }
-  case object Introductory extends FrontendId { val name = "Introductory" }
-  case object SixWeeks extends FrontendId { val name = "SixWeeks" }
-
-  val all = List(OneYear, ThreeMonths, Monthly, Quarterly, Yearly, Introductory, SixWeeks)
-
-  def get(jsonString: String): Option[FrontendId] =
-    all.find(_.name == jsonString)
+  def totalChargesMinorUnit: Int =
+    ratePlanCharges.map(c => (c.pricing.prices.head.amount * 100).toInt).list.toList.sum
+
+  def chargesPrice: PricingSummary =
+    ratePlanCharges
+      .map(_.pricing)
+      .list
+      .toList
+      .reduce((f1, f2) =>
+        PricingSummary(
+          f1.underlying.keySet
+            .intersect(f2.underlying.keySet)
+            .map { currency =>
+              currency -> f1.underlying(currency).+(f2.underlying(currency))
+            }
+            .toMap,
+        ),
+      )
+
+  def billingPeriod: Validation[String, BillingPeriod] = {
+    val billingPeriods = ratePlanCharges.list.toList.map(c => c.billingPeriod).distinct
+    billingPeriods match {
+      case Nil => Validation.f[BillingPeriod]("No billing period found")
+      case b :: Nil => b
+      case _ => Validation.f[BillingPeriod]("Too many billing periods found")
```

Comment:
I checked bigquery and there were only 5 active subscriptions with rateplans containing mixed billing periods, all were quarterly migrated in echo legacy subscriptions (the monthly charges were all zero)
before it would have been incorrectly hard coded as monthly, now it will end up as unknown_billing_period.
I did consider filtering the free charges out, but I found there are quite a lot of free charges in the system, mainly recurring contributions that have been set to zero, but a few digi sub and others.